### PR TITLE
matcher: remove WIP annotation.

### PR DESCRIPTION
**Pull request by tyxia**
Commit Message: remove WIP annotation. unified matcher has been used in production for quite a while

### CODE REVIEW

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!
/lgtm api

File: api/envoy/config/route/v3/route_components.proto
Comment (alyssawilk) [inline api/envoy/config/route/v3/route_components.proto:96]: Given we don't expect a next major version should we clean this up too?

**Review by alyssawilk — APPROVED** [review]

File: api/envoy/config/route/v3/route_components.proto
Comment (adisuissa) [reply api/envoy/config/route/v3/route_components.proto:96]: I think this can be removed, but should be replaced with oneof-promotion tag. Not because it will ever change, but because it helps detecting all those fields that should be oneofs.

Somewhat related, I'd like to gradually change the route data-structure that we are using at the moment, so instead of including all the fields of a oneof in the same level in the data-structure, refactor them into different implementations of some interface (using the Strategy pattern). This should save some memory, and hopefully make the code a bit cleaner.

**Comment by alyssawilk** [comment]
Ah, looks like format fail is real
/wait

File: api/envoy/config/route/v3/route_components.proto
Comment (tyxia) [reply api/envoy/config/route/v3/route_components.proto]: yep.  `oneof_promotion` is standard approach for this case. It probably should be done in this way when `matcher` was introduced. 

Done.

**Review by alyssawilk — COMMENTED** [review]
still LGTM from a senior maintainer perspective, pending api sign-off

**Review by adisuissa — COMMENTED** [review]
/lgtm api

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**
